File: app/src/main/java/space/pal/sig/view/news/NewsViewModel.java
package space.pal.sig.view.news;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.paging.PagedList;

import space.pal.sig.model.News;
import space.pal.sig.repository.NewsRepository;

/**
 * SpaceNow
 * Created by Catalin on 7/19/2020
 **/
public class NewsViewModel extends AndroidViewModel {

    private final NewsRepository newsRepository;
    private final LiveData<PagedList<News>> newsList;

    public NewsViewModel(@NonNull Application application, NewsRepository newsRepository) {
        super(application);
        this.newsRepository = newsRepository;
        this.newsList = newsRepository.findAll(30);
    }

    public LiveData<PagedList<News>> getNewsList() {
        return newsList;
    }
}

File: app/src/main/java/space/pal/sig/view/news/NewsViewModel.kt
package space.pal.sig.view.news

import android.app.Application
import androidx.lifecycle.LiveData
import space.pal.sig.model.entity.News
import space.pal.sig.repository.NewsRepository
import space.pal.sig.view.BaseViewModel

/**
 * SpaceNow
 * Created by Catalin on 12/25/2020
 **/
class NewsViewModel(
        application: Application,
        newsRepository: NewsRepository
) : BaseViewModel(application) {

    private val newsList = newsRepository.findAll()

    fun getNewsList(): LiveData<MutableList<News>> {
        return newsList
    }

}